connectors/vault/vaultconn-bl/vault_reader.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package vaultconnbl
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	vaultutils "github.com/ibm/the-mesh-for-data/connectors/vault/vault_utils"
+	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
+)
+
+type Server struct {
+}
+
+func (s *Server) GetCredentialsInfo(in *pb.DatasetCredentialsRequest, vault vaultutils.VaultConnection, vaultPathKey string) (*pb.DatasetCredentials, error) {
+	log.Println("Vault connector: GetCredentialsInfo")
+	log.Printf("Vault connector: GetCredentialsInfo, in = %v\n", in)
+
+	// secretaddr := url.QueryEscape(in.DatasetId)
+	secretaddr := strings.ReplaceAll(in.DatasetId, " ", "")
```

Comment:
Removing all spaces everywhere seems a bit overkill. Can you please give me some background here over how the dataset ids look like and why this needs to be done? If you just want to remove spaces in begin and end of the string there is also `strings.TrimSpace`. If the ids are sensitive you can also contact me on Slack.

Also please remove the commented code above if not needed.

Reply:
The in.DatasetId will be of [this form](https://github.com/IBM/the-mesh-for-data/blob/641d199fa1ed8505b520580dadbe5666b4ef517d/connectors/vault/user_vault_init.sh#L29). In case there is any space we are removing it. It is more of a safety statement. I have removed the comment above in Line 22 as you mentioned.